buildstrap

```diff
@@ -11,18 +11,42 @@ OUT_TMP=$4
 PACKAGES=$5
 INSTALL_BCC=$6
 MIRROR=$7
+DISTRO_VERSION=$8
 
-root=sudo
+if [[ ! -z $DISTRO_VERSION ]]; then
+    # Assume non-empty DISTRO_VERSION is Ubuntu; see if there's a base image..
+    echo "Checking for Ubuntu Base image from the web..."; echo ""
+    if [ ! -f $TDIR/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz ]; then
+        curl -L http://cdimage.ubuntu.com/ubuntu-base/releases/$DISTRO_VERSION/release/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz \
+            --output $TDIR/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz; fi
+    if [ -f $TDIR/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz ]; then
+        time tar -C $OUT_TMP --exclude=dev -xf $TDIR/ubuntu-base-$DISTRO_VERSION-base-$ARCH.tar.gz;
+        mkdir $OUT_TMP/dev; fi; fi
 
-if [ $EUID -ne 0 ]; then echo "The next stage runs as sudo, please enter password if asked."; fi
+if [ ! -f $OUT_TMP/etc/debian_version ]; then
```

Comment:
Above, you are using "non-empty DISTRO_VERSION is Ubuntu" but here, you're relying on debian_version for the same. Lets keep it consistent?
Also, can  we instead do something like "check for the word ubuntu in the $MIRROR", and if so then its ubuntu? That's much less fragile to me than the other methods. What do you think?